src/Core/Grid/Query/CategoryQueryBuilder.php

```diff
@@ -171,6 +171,16 @@ private function getQueryBuilder(array $filters)
             }
 
             if ('position' === $filterName) {
+                // When filtering by position,
+                // value must be decreased by 1,
+                // since position value in database starts at 0,
+                // but for user display positions are insreased by 1.
+                if (is_numeric($filterValue)) {
+                    --$filterValue;
```

Comment:
What happens if `$filterValue` is 0 😉?

We cannot trust browser input 😛

Reply:
I think it doesn't matter. We don't perform any write operations, so you are free to enter even -999, you just won't get any results. 

Correct me if I'm missing something. :thinking:

Reply:
Isn't it a broken window ? 😉 "The code lets you input anything irrelevant as it cannot harm the application because the database is correct"

Reply:
Hmm, I don't see how it would be. :thinking: It cannot harm application because it's a read model. 

For example, if we have products grid with prices from $10 to $100. Does that mean user cannot enter $110 to filter by it?

I don't think this is something that needs to be fixed, as this behavior is also available on other grids and does not cause any issues. Wdyt?

Reply:
Make sense 👍